### ISSUE TITLE
Plugin prevents creation of Moodle assignment if having connection issues

### ISSUE THREAD

**Issue by mhughes2k** [turnitin/moodle-plagiarism_turnitin#626]
We've started to get these errors:

![image](https://user-images.githubusercontent.com/694841/175520295-9e2c5bd6-0160-4db8-a49e-dc98b5a786f7.png)

This is upon clicking the "create assignment" option in the Moodle activity chooser, the actual Moodle form has not been displayed yet.

It is unacceptable that the plagiarism plugin interferes with standard Moodle processes in this way. Any connection issues with Turnitin should be logged, and then the plugin *disabled* for this process, it should *not* fail the whole thing.

**Comment by mhughes2k** [comment]
Stack trace:
line 494 of /lib/setuplib.php: moodle_exception thrown
line 3369 of /plagiarism/turnitin/lib.php: call to print_error()
line 151 of /plagiarism/turnitin/classes/turnitin_comms.class.php: call to plagiarism_turnitin_print_error()
line 129 of /plagiarism/turnitin/classes/turnitin_assignment.class.php: call to turnitin_comms->handle_exceptions()
line 1662 of /plagiarism/turnitin/lib.php: call to turnitin_assignment->create_tii_course()
line 625 of /plagiarism/turnitin/lib.php: call to plagiarism_plugin_turnitin->create_tii_course()
line 328 of /plagiarism/turnitin/lib.php: call to plagiarism_plugin_turnitin->get_course_data()
line 117 of /lib/plagiarismlib.php: call to plagiarism_plugin_turnitin->get_form_elements_module()
line 192 of /mod/assign/mod_form.php: call to plagiarism_get_form_elements_module()
line 214 of /lib/formslib.php: call to mod_assign_mod_form->definition()
line 130 of /course/moodleform_mod.php: call to moodleform->__construct()
line 141 of /course/modedit.php: call to moodleform_mod->__construct()

**Comment by mhughes2k** [comment]
I'm sorry but *why* is Turnitin doing *anything* involving a live connection during the construction of a moodle form, without catching exceptions this could raise.

**Comment by dwinn** [comment]
Hi @mhughes2k ,

Which version of the plugin are you using?

We fixed this in 2021060801 and I just double checked and can no longer reproduce this issue. The release note for that version that mentions this is as follows:

_Create or access Moodle assignments, forums, and workshops when Turnitin is disabled_
If Turnitin is disabled for an assignment, forum, or workshop and Turnitin is unavailable by having the wrong configuration options or being under maintenance, then an error message would be shown. This bug has been fixed, and Turnitin being unreachable will no longer impact your ability to create or access your content.
